### PR TITLE
CI: Add central 'go test' script

### LINKED ISSUE #16: Create a central go test script
Following the pattern of previous PRs, we can create a central `go test` script that can be used by all the repositories.

### PR DESCRIPTION
**Pull request by jodh-intel**
Add a generic 'go test' unit testing script that can be called
by the other repositories.

Fixes #16.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Script is based on https://github.com/clearcontainers/runtime/blob/master/.ci/go-test.sh.

**Comment by jodh-intel** [comment]
Hi @chavafg, @bergwolf - could you take a look please?

**Comment by chavafg** [comment]
lgtm

File: .ci/go-test.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	echo "INFO: Currently running as user '$(id -un)'"
+
+	for pkg in $test_packages; do
+		for user in $users; do
+			printf "INFO: Running 'go test' as %s user on package '%s' with flags '%s'\n" \
+				"$user" "$test_packages" "$go_test_flags"
+
+			run_as_user "$user" go test $go_test_flags -covermode=atomic -coverprofile="$tmp_coverage_file" "$pkg"
+
+			if [ -f "${tmp_coverage_file}" ]; then
```
Comment (grahamwhaley) [inline .ci/go-test.sh]: This line feels oddly redundant, but benign, to me - maybe the mix of `tmp` and `test` prefixes is confusing me - won't this file always exist from the line above it? I suspect I'm missing something subtle?, in which case, do we think we can elaborate a touch in the script?

Comment (jodh-intel) [reply .ci/go-test.sh]: That's a good question, but the test is in fact required. I've added a comment explaining why (and also caught and fixed a bug in the `printf` call above by doing that).

**Pull request merged**